specification/azurefleet/AzureFleet.Management/fleet.tsp

```diff
@@ -0,0 +1,228 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-autorest";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Autorest;
+using Azure.ResourceManager;
+using OpenAPI;
+
+namespace Microsoft.AzureFleet;
+
+@doc("An Compute Fleet resource")
+model Fleet is TrackedResource<FleetProperties> {
+  @doc("The name of the Compute Fleet")
+  @key("fleetName")
+  @segment("fleets")
+  @pattern("^[^_\\W][\\w-._]{0,79}(?<![-.])$")
+  @path
+  name: string;
+
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "Zones is a valid property"
+  @doc("Zones in which the Compute Fleet is available")
+  zones?: Array<string>;
+
+  ...ManagedServiceIdentityProperty;
+  ...ResourcePlanProperty;
+}
+
+@doc("Fleet Update Model")
+model FleetUpdate {
+  ...Azure.ResourceManager.Foundations.ArmTagsProperty;
+
+  /** Updatable managed service identity */
+  identity?: ManagedServiceIdentityUpdate;
+
+  /** Updatable resource plan */
+  plan?: ResourcePlanUpdate;
+
+  /** RP-specific updatable properties */
+  properties?: FleetProperties;
+}
+
+model ManagedServiceIdentityUpdate
+  is OptionalProperties<UpdateableProperties<Azure.ResourceManager.Foundations.ManagedServiceIdentity>>;
+
+model ResourcePlanUpdate
+  is OptionalProperties<UpdateableProperties<Azure.ResourceManager.Foundations.Plan>>;
+
+@doc("The status of the current operation.")
+@Azure.Core.lroStatus
+union ProvisioningState {
```

Comment:
>ProvisioningState [](http://example.com/codeflow?start=6&length=17)

I think you can include the built-in ResourceProvisioningState (for the standard/required states) and just add your additional states here. Just something I noticed, not required or anything.

Reply:
Updated.